src/externalTypes.ts

```diff
@@ -0,0 +1,27 @@
+interface SharedOpts {
+  /** JS target */
+  target: 'node' | 'browser';
+  /** Path to tsconfig file */
+  tsconfig?: string;
+  /** Is error extraction running? */
+  extractErrors?: boolean;
+}
+
+type ModuleFormat = 'cjs' | 'umd' | 'esm' | 'system';
+
+export interface TsdxOptions extends SharedOpts {
+  /** Name of package */
+  name: string;
+  /** path to file */
+  input: string;
+  /** Environment */
+  env: 'development' | 'production';
+  /** Module format */
+  format: ModuleFormat;
+  /** Is minifying? */
+  minify?: boolean;
+  /** Is this the very first rollup config (and thus should one-off metadata be extracted)? */
+  writeMeta?: boolean;
+  /** Only transpile, do not type check (makes compilation faster) */
+  transpileOnly?: boolean;
+}
```

Comment:
```suggestion
export { TsdxOptions } from './types'
```

Is there a reason you didn't just re-export that one type?

Otherwise, if just creating a single declaration file, the preference would be to just use a `.d.ts` file and not add any other declarations

Reply:
If re-export from './types'. User can explore more types which you don't want show public. e.g. Cmd+Click in vscode.

Reply:
Yes but they can do that anyway if `declaration` is on. Nothing else is exported from `tsdx`, the rest have to be pulled from `tsdx/dist/` which is an undocumented and implicitly private API, so "use at your own risk" is implied there.